Framework/Core/include/Framework/ASoA.h

```diff
@@ -1988,6 +1990,77 @@ std::tuple<typename Cs::type...> getRowData(arrow::Table* table, T rowIterator,
 {
   return std::make_tuple(getSingleRowData<T, Cs>(table, rowIterator, ci, ai, globalIndex)...);
 }
+
+template <typename R, typename T, typename C>
+R getColumnValue(const T& rowIterator)
+{
+  return static_cast<R>(static_cast<C>(rowIterator).get());
+}
+
+template <typename R, typename T>
+using ColumnGetterFunction = R (*)(const T&);
+
+template <typename T, typename R>
+concept dynamic_with_common_getter = is_dynamic_v<T> &&
+                                     // lambda is callable without additional free args
+                                     framework::pack_size(typename T::bindings_t{}) == framework::pack_size(typename T::callable_t::args{}) &&
+                                     requires(T t) {
+                                       { t.get() } -> std::convertible_to<R>;
+                                     };
+
+template <typename T, typename R>
+concept persistent_with_common_getter = is_persistent_v<T> && requires(T t) {
+  { t.get() } -> std::convertible_to<R>;
+};
+
+template <typename R, typename T, persistent_with_common_getter<R> C>
+ColumnGetterFunction<R, T> createGetterPtr(const std::string_view& columnLabel)
+{
+  const size_t n = columnLabel.size();
+
+  if (n == 0 || n != strlen(C::columnLabel())) {
+    return nullptr;
+  }
+
+  return (std::strcmp(columnLabel.data(), C::columnLabel())) ? nullptr : &getColumnValue<R, T, C>;
+}
+
+template <typename R, typename T, dynamic_with_common_getter<R> C>
+ColumnGetterFunction<R, T> createGetterPtr(const std::string_view& columnLabel)
+{
+  const size_t n = columnLabel.size();
+
+  if (n == 0 || (n != strlen(C::columnLabel()) && n - 1 != strlen(C::columnLabel()))) {
+    return nullptr;
+  }
+
+  return ((std::strcmp(&columnLabel[1], C::columnLabel()) && std::strcmp(columnLabel.data(), C::columnLabel()))) ? nullptr : &getColumnValue<R, T, C>;
```

Comment:
I simplified it a bit. What do you think?

Reply:
I made some tests and spotted a bug; using strncmp like that will not work without a length check. Edge case:
columnLabel = "fP"
C::columnLabel = "fPx"
With strncmp with n = columnLabel.size() and if "fPx" occurs in an iteration before "fP", the wrong thing is returned.
Because of that, I decided to use the `==` operator between `string_view` and `const char*`. Performance should be similar to these complicated checks, and the code should have better readability. I hope this solution is sufficient!